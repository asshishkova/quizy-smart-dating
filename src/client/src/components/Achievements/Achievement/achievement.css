.achievement-container{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    background-color: white;
    width: 80%;
    border-radius: 20px;
    padding: 20px;
}

.achievement-chip{
    min-width: 30%;
}

