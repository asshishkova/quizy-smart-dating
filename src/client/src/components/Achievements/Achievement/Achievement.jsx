import "./achievement.css";
import { Chip, Avatar } from "@mui/material";
import Movies from "../../../images/movies.png";

export default function Achievement({ title, score }) {
  const getScoreFromArray = () => {
    const scored = Math.min(...score);
    const answerdTotal = Math.max(...score);
    return `scored ${scored} out of ${answerdTotal}`;
  };

  return (
    <div className="achievement-container">
      <Chip
        className="achievement-chip"
        label={title}
        avatar={<Avatar alt="Natacha" src={Movies} />}
      />
      <p>{getScoreFromArray()}</p>
    </div>
  );
}
