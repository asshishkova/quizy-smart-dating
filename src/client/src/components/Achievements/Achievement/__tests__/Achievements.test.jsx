import renderer from "react-test-renderer";
import Achievement from "../Achievement";

const achievement = { title: "Film", score: {correct: 8, answers: 10}}

describe("Achievement component", () => {
  test('renders achievement correctly', () => {
    const tree = renderer.create(
      <Achievement title={achievement.title} score={achievement.score} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
})
