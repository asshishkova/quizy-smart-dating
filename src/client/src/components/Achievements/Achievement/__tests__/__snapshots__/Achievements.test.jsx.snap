// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders achievement correctly 1`] = `
<div
  className="achievement-container"
>
  <div
    className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorError MuiChip-filledError achievement-chip css-13vwgur-MuiChip-root"
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    style={
      Object {
        "boxShadow": "2px 2px 3px gray",
        "display": "flex",
        "justifyContent": "flex-start",
      }
    }
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorError css-i4bv87-MuiSvgIcon-root"
      data-testid="MovieCreationIcon"
      focusable="false"
      style={
        Object {
          "color": "#111111",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="m18 4 2 4h-3l-2-4h-2l2 4h-3l-2-4H8l2 4H7L5 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V4h-4z"
      />
    </svg>
    <span
      className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
    >
      Film
    </span>
  </div>
  <div
    className="achievement-text"
  >
    <p
      className="precent"
    >
      80
      % correct
    </p>
    <p
      className="number"
    >
      (
      8
       out of 
      10
      )
    </p>
  </div>
  <div
    className="achievement-pie"
  >
    <svg
      height="100%"
      viewBox="0 0 200 100"
      width="100%"
    >
      <path
        d="M 75 50 A 25 25 0 1 1 92.27457514062633 73.77641290737884"
        fill="none"
        stroke="#d32f2f"
        strokeWidth={50}
        style={Object {}}
      >
        <title>
          Correct
        </title>
      </path>
      <path
        d="M 92.27457514062633 73.77641290737884 A 25 25 0 0 1 75 50.00000000000001"
        fill="none"
        stroke="#CCCCCC"
        strokeWidth={50}
        style={Object {}}
      >
        <title>
          Incorrect
        </title>
      </path>
    </svg>
  </div>
</div>
`;
